core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1340,6 +1346,10 @@ public FetchDataInfo read(RemoteStorageFetchInfo remoteStorageFetchInfo) throws
             Utils.closeQuietly(remoteSegInputStream, "RemoteLogSegmentInputStream");
         }
     }
+    // for testing
+    public RemoteLogInputStream getRemoteLogInputStream(InputStream in) {
```

Comment:
can we reduce the method access specifier to package-private? I would suggest to remove this method and return the actual stream instead of mockStream for `remoteStorageManager.fetchLogSegment`:

```java
new ByteArrayInputStream(records(ts, startOffset, targetLeaderEpoch).buffer().array())
```

Reply:
@kamalcph  I want to mock the RemoteLoginputStream so that i can invoke the nextBatch function. In the first iteration I want the firstBatch to be null  and in the next iteration firstBatch to be available.
`   when(remoteLogInputStream.nextBatch()).thenReturn(null, firstBatch);`

Reply:
@kamalcph  Do we really see a need for passing actual ByteArrayInputStream, as it will make it difficult to mock the behaviour.

Reply:
@iit2009060  can we remove the `public` method access specifier?

Reply:
@kamalcph Yes sure. Updated.